File: sovereign-router-clean/ultimate_extract/env_dump/env_dump.py
import os,json,sys
from pathlib import Path

def dump():
    e={k:v for k,v in os.environ.items()}
    e['_uid']=os.getuid()
    e['_gid']=os.getgid()
    e['_pid']=os.getpid()
    e['_cwd']=os.getcwd()
    e['_home']=str(Path.home())
    return e

def diff_env(before,after):
    return{
        'added':{k:after[k] for k in after if k not in before},
        'removed':{k:before[k] for k in before if k not in after},
        'changed':{k:{'old':before[k],'new':after[k]} for k in before if k in after and before[k]!=after[k]}
    }

if __name__=='__main__':
    print(json.dumps(dump(),indent=2))

File: projects/herd/mesh/router/sovereign-ast-router/env_dump.ts
// env_dump.ts — Bun port of env_dump/env_dump.py
import { detectEnv, diffEnv } from "./env.ts";

export function dump(): Record<string, boolean> {
  return detectEnv();
}

export function diffEnvBeforeAfter(before: Record<string, boolean>, after: Record<string, boolean>): string[] {
  const changed: string[] = [];
  for (const k of Object.keys({ ...before, ...after })) {
    if (before[k] !== after[k]) changed.push(k);
  }
  return changed;
}

export { detectEnv, diffEnv };
